code/scripting/api/libs/mission.cpp

```diff
@@ -1510,6 +1510,28 @@ ADE_FUNC(getArrivalList,
 	                    l_ParseObject.Set(parse_object_h(&Ship_arrival_list)));
 }
 
+ADE_FUNC(getShipList,
+	l_Mission,
+	nullptr,
+	"Get an iterator to the list of ships in this mission",
+	"iterator<ship>",
+	"An iterator across all ships in the mission. Can be used in a for .. in loop")
+{
+	ship_obj* so = &Ship_obj_list;
+
+	return ade_set_args(L, "u", luacpp::LuaFunction::createFromStdFunction(L, [so](lua_State* LInner, const luacpp::LuaValueList& /*params*/) mutable -> luacpp::LuaValueList {
+		//Since the first element of a list is the next element from the head, and we start this function with the head as so, this GET_NEXT will return the first element on first call of this lambda.
```

Comment:
The comment is good, but reword "as so".  I know you're referring to `so` the variable, but "so" is an actual word and "as so" can be misinterpreted as a comparison.